base/array.jl

```diff
@@ -827,6 +836,18 @@ function deleteat!(a::Vector, inds)
     return a
 end
 
+function deleteat!(a::Vector, inds::AbstractVector{Bool})
```

Comment:
Sounds interesting, but I discovered several issues when trying this:
- The fact that `to_indices` does not support arbitrary iterables would introduce a regression here: for example, `deleteat!([1,2,3], (i for i in 2:3))` is currently supported. Or maybe support for this could be added to `to_indices`? Else, we could call `to_indices` only for `AbstractVector` indices, and keep the existing general definition in other cases.

- The above `AbstractVector{Bool}` method is very simple and efficient, and the `LogicalIndex` type returned by `to_indices` makes it actually more complex. This is a case where integer indices are less useful than boolean indices. Again, one solution is to keep the specialized `AbstractVector{Bool}` in parallel with the more general framework.

I've updated the PR to use `to_indices`, but keeping both the most general (iterators) and the specialized `AbstractArray{Bool}` signature. Tell me what you think.